Task01/Var1

statizarm opened the pull request.

statizarm committed:
Merge branch 'main' of github.com:xthebat/iu4-2k21-mlni

statizarm committed:
Конспект семинара
- Сохранаяю конспект семинара

statizarm committed:
Merge branch 'main' of github.com:xthebat/iu4-2k21-mlni

statizarm committed:
task01
- Написал функцию загружающую из файла лабиринт
- Добавил .gitignore для игнора venv и .idea

statizarm committed:
task01
- Добавил функционал созадния матрицы смежности
- Изменил логику построения
-- До: чтение из файла карты лабиринта, из которой создается граф
-- После: построчное чтение файла с непосредственным созданием графа

xthebat commented on shushuev_ai/task01/main.py:
```diff
@@ -0,0 +1,58 @@
+
+
+def main(args):
+    filename = args[1] if len(args) > 1 else input("Enter path to csv file: ")
```
Не используйте input - я говорил об этом на семинарах. Это неудобно для тестирования и бессмысленно (в таком варианте консольные программы практически никогда не используются)

xthebat commented on shushuev_ai/task01/main.py:
```diff
@@ -0,0 +1,58 @@
+
+def generate_vertexes_from_file(filepath):
+    file = open(filepath, "rt")
+    row_gen = ([int(num) for num in line.split(' ')] for line in file)
```
Отдельно можно не создавать генератор, а сразу писать его в if

xthebat commented on shushuev_ai/task01/main.py:
```diff
@@ -0,0 +1,58 @@
+
+
+def generate_vertexes_from_file(filepath):
+    file = open(filepath, "rt")
```
Не оставляйте файлы открытыми - используйте with

xthebat commented on shushuev_ai/task01/main.py:
```diff
@@ -0,0 +1,58 @@
+    vid = 1
+    for row in row_gen:
+        prev = 0
+        for i in range(len(row)):
```
наверное i лучше назвать col :)

xthebat commented on shushuev_ai/task01/main.py:
```diff
@@ -0,0 +1,58 @@
+                row[i] = str(vid)
+                vid += 1
+
+                if prev == 1:
```
Логика работы с prev - непонятна :( Не говорю, что не правильно или это плохо, в комментариях хотел бы описание как это работает. На мой взгляд нужно тыкаться в каждого соседа на каждой ячейки, которая является "проходом". См. https://github.com/xthebat/iu4-2k21-mlni/pull/12#pullrequestreview-613945389

statizarm replied:
я изменил в новой версии алгоритм построения, теперь там нет prev. Но, в этой версии он нужен был как ограничитель и переменная которая хранит в себе значение предыдущей ячейки в строке (для 0й чейки значение предыдущей ячейке в этой же строке должно быть 0)

statizarm replied:
Если prev убрать, то для того чтобы посмотреть предыдущее значение нужно было бы проверять i(col) на 0, чтобы не вылететь за границы массива

xthebat reviewed (changes requested):
Нужны:
- комментарии как работает логика с prev
- unit-test
- "оптимизация" графа

statizarm committed:
task01
- Заменил предыдущую версию генерации графа на новую:
-- Идея заключается в использовании волнового алгоритма для обхода
лабиринта и если вершина имеет отличное 1 колическо непосещенных еще
соседей она заносится в матрицу смежности
- Добавил тесты:
-- Тест на чтение лабиринта из файла
-- Тест на построение графа для простого случая
-- Тест на построение графа для лабиринта с островом
-- Тест на построение графа для лабиринта с несколькими узлами
-- Тест на построение графа для лабиринта с циклическим путем

statizarm committed:
task01
- Добавил построение графа при обходе лабиринта волновым алгоритмом с
  возможностью сноса 1 стены
- Добавил тесты проверяющие функционал сноса стены
-- Тест в лабиринте без пути от начала до конца
-- Тест в лабиринте где снос стены дает уменьшение стоимости пути

Commit:
CR-Task01

statizarm committed:
task01
- Вмержил изменения из CR
- Добавил класс для точки
- Добавил класс для лабиринта
- Добавил класс для состояния волнового алгоритма
- Зарефакторил функцию генерации точек расположения стен, которые можно сломать
- Зарефакторил функцию строющую волновым алгоритмом матрицу смежности

statizarm committed:
task01
- Вынес остатки итерций волнового алгоритма в __next__ метод класса State

statizarm committed:
task01
- Подрефакторил согласно CR
